integrations/access/accessrequest/app.go

```diff
@@ -185,6 +222,38 @@ func (a *App) onWatcherEvent(ctx context.Context, event types.Event) error {
 	}
 }
 
+func (a *App) handleAccessMonitoringRule(ctx context.Context, event types.Event) error {
+	if kind := event.Resource.GetKind(); kind != types.KindAccessMonitoringRule {
+		return trace.BadParameter("expected %s resource kind, got %s", types.KindAccessMonitoringRule, kind)
+	}
+
+	e, ok := event.Resource.(types.Resource153Unwrapper)
+	if !ok {
+		return trace.BadParameter("expected AccessMonitoringRule resource type, got %T", event.Resource)
+	}
+	req, ok := e.Unwrap().(*accessmonitoringrulesv1.AccessMonitoringRule)
+	if !ok {
+		return trace.BadParameter("expected AccessMonitoringRule resource type, got %T", event.Resource)
+	}
+
+	if !a.amrAppliesToThisPlugin(req) {
+		return nil
+	}
+
+	a.accessMonitoringRules.Lock()
+	defer a.accessMonitoringRules.Unlock()
+	switch op := event.Type; op {
+	case types.OpPut:
+		a.accessMonitoringRules.rules[req.Metadata.Name] = req
+		return nil
+	case types.OpDelete:
+		delete(a.accessMonitoringRules.rules, req.Metadata.Name)
+		return nil
+	default:
+		return trace.BadParameter("unexpected event operation %s", op)
+	}
```

Comment:
If OP is delete, the resource can be:
- types.ResourceHeader
- types.Resource153Unwrapper

This happens because if expiration happens in DB, Teleport doesn't have the full payload and reports only kind, name...

Same for access requests but in that case, the operation doesn't enforce type check prior to decide the operation type